--- ffi-1.15.5/lib/ffi/platform.rb.orig	2023-02-18 19:37:47.086493199 +0100
+++ ffi-1.15.5/lib/ffi/platform.rb	2023-02-18 19:38:48.586860130 +0100
@@ -106,7 +106,7 @@ module FFI
     name_version = "12" if IS_FREEBSD && OSVERSION >= 12 # 64-bit inodes
 
     NAME = "#{ARCH}-#{OS}#{name_version}"
-    CONF_DIR = File.join(File.dirname(__FILE__), 'platform', NAME)
+    CONF_DIR = File.join(RbConfig::CONFIG["vendorarchdir"], 'ffi', 'platform', NAME)
 
     public
 
